### PR TITLE
Move scheduling and deletion into `...` menu

### LINKED ISSUE #820: Move scheduling and deletion to `…` menu
follow up of #767

### LINKED ISSUE #846: vue runtime warning
`[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup().`

Bildschirmfoto vom 2024-09-24 08-37-34

### Steps to reproduce
1. Open developer console
2. go into announcement app
3. Done

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration

**App Version**: 7.0.0

**Web Browser:** Firefox

### PR DESCRIPTION
**Pull request by mwinkens**
closes #820
closes #846 (for some reason)

This was the best I was able to come up with for now. I think it's not too bad with the separators.
@nickvergessen I don't know how to handle the translations, I would remove the `(optional)` part here.

Bildschirmfoto vom 2024-09-24 15-14-12

### CODE REVIEW

**Comment by nickvergessen** [comment]
> I don't know how to handle the translations, I would remove the (optional) part here.

Sounds good

File: src/Components/NewForm.vue [hunk 1]
```diff
@@ -96,6 +71,32 @@
 					:checked.sync="allowComments">
 					{{ t('announcementcenter', 'Allow comments') }}
 				</NcActionCheckbox>
+				<NcActionSeparator />
+				<NcActionCheckbox value="0"
+					:checked.sync="scheduleEnabled">
+					{{ t('announcementcenter', 'Schedule announcement time (optional)') }}
+				</NcActionCheckbox>
+				<NcDateTimePicker v-model="scheduleTime"
```
Comment (nickvergessen) [inline src/Components/NewForm.vue:79]: Handling of this feels quite complicated and e.g. there is no "easy" way to close it after you selected minutes.
Maybe we can do something closer to the reminder in Talk?
https://github.com/nextcloud/spreed/blob/main/src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue#L192-L200

Comment (mwinkens) [reply src/Components/NewForm.vue]: @nickvergessen I started implementing this, while this works as well I ran into a few issues:
- you need to select the hours or this component doesn't default to 0:00, which feels un-intuitive
- I can't get rid of that "Please choose a date" text, I updated the screenshot in the description

**Comment by nickvergessen** [comment]
> I can't get rid of that "Please choose a date" text, I updated the screenshot in the description

Managed to do that

**Comment by nickvergessen** [comment]
> you need to select the hours or this component doesn't default to 0:00, which feels un-intuitive

Seems to only be a problem with Firefox, Chrome shows a time picker next to it and automatically sets the current time:

Firefox | Chrome
---|---
grafik | grafik

**Review by nickvergessen — APPROVED** [review]

**Comment by JGHobbyist** [comment]
This is such a great idea... However when setting a scheduled announcement, I pick the date and time and get the following error... This then doesn't send the announcement out at all (by email or notification) but only keeps it in the center... Is there something that I need to configure that I may have missed? 

Thank you for all the work you do in making Nextcloud better!
Test Schedule Announcement Before


Test Schedule Announcement After

**Comment by nickvergessen** [comment]
- It's just wrong text until the time is reached: https://github.com/nextcloud/announcementcenter/pull/848

Fixed already, just waiting for a release

**Comment by JGHobbyist** [comment]
That's fantastic, Thank you. I asked because the announcement didn't get sent at all when the schedule option was used. So I guess the next release this will send the scheduled announcement? 

"just waiting for a release" - as in you're waiting for the next NC update release or you're waiting for the next Announcement center release to group the fixes together? 

Thanks again.

**Comment by nickvergessen** [comment]
The later

**Comment by nickvergessen** [comment]
Released 7.0.1 btw

**Comment by Jerome-Herbinet** [comment]
Hi @nickvergessen 

The “Schedule announcement time” and “Schedule deletion time” entries are no longer visible, but they are still present in the source code. 

Was this PR the cause of this? 



I actually think that icons alone are insufficient in terms of user interface and user experience for users. I suggest reintroducing the labels mentioned.

Thanks

**Comment by Jerome-Herbinet** [comment]
> Hi @nickvergessen
> 
> The “Schedule announcement time” and “Schedule deletion time” entries are no longer visible, but they are still present in the source code.
> 
> Was this PR the cause of this?
> 
> 
> I actually think that icons alone are insufficient in terms of user interface and user experience for users. I suggest reintroducing the labels mentioned.
> 
> Thanks

Hi @nickvergessen can you just answer ; it will help me to better understand things. Thanks 🙂

**Comment by mwinkens** [comment]
Dear @Jerome-Herbinet , I'd open a separate issue for this. I am not sure you want to ping everyone here and issues are handled a bit different than comments

**Pull request merged**